### PR TITLE
Submitting solution and tests for Ticket 1

### PR DESCRIPTION
**Pull request by AnnaTas77**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+});
+
+describe("GET /api/topics", () => {
+    test("200: should respond with an array of all topic objects", () => {
+        return request(app)
+            .get("/api/topics")
+            .expect(200)
+            .then(({ body }) => {
+                const { topics } = body;
+                expect(topics).toBeInstanceOf(Array);
```
Comment (RichardNeat) [inline __tests__/app.test.js:22]: Would be ideal to see a length check here, if the response came back as an empty array this expect statement would pass and the forEach would never run giving you an auto pass. So we need to guard against this with a length check to be certain there is content in there and the forEach will run.

File: errors/errors.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+        res.status(400).send({ msg: "Bad request" });
+    } else next(err);
+};
+
+exports.handleCustomErrors = (err, req, res, next) => {
+    if (err.msg) {
+        res.status(err.status).send({ msg: err.msg });
+    } else next(err);
+};
+
```
Comment (RichardNeat) [inline errors/errors.js]: We currently have no need for any PSQL or custom error handlers so we shouldn't include this yet.

**Review by RichardNeat — CHANGES_REQUESTED** [review]
Nice one Anna - just a couple of changes requested below to do please.
It might also be nice to include a generic 404 bad path handler at this stage too.
Please make these changes and then resubmit. Thanks :)

File: __tests__/app.test.js [hunk 3]
```diff
@@ -20,6 +20,7 @@ describe("GET /api/topics", () => {
             .then(({ body }) => {
                 const { topics } = body;
                 expect(topics).toBeInstanceOf(Array);
+                expect(topics).toHaveLength(3);
```
Comment (Dutton1990) [inline __tests__/app.test.js:23]: Nice job!

**Comment by Dutton1990** [comment]
Nice job Anna! We will cover in the lecture tomorrow morning how to handle a bad path so you can implement that tomorrow. 

You can merge this ticket and continue now, thanks for resubmitting!

**Pull request merged**